### DISCUSSION TITLE
Vaultwarden  command line client in Rust ?

### DISCUSSION THREAD

**Discussion by tjyang** [dani-garcia/vaultwarden#2861] [Ideas]
- WHAT : I  like to see VW has its own command line client written in Rust.
- WHY:    Using VW own client in rust to test its own server.  VW test suite not done yet.
- HOW:   Archived [bitwarden-cli-rust ](https://github.com/birlorg/bitwarden-cli/tree/trunk/rust) could be good starting point for those  like to learn Rust by doing.

**Comment by BlackDex** [comment]
While it might be fun and educational for someone who wants to learn rust, I'm not sure of it is that useful. The official clients work on all environments, and Vaultwarden i try to stay compatible with the latest clients.

But, everyone is free to build this of course, but i do not think one of the main maintainers will start on such an endeavor though.

**Reply by tjyang** [reply]
Sure, no problem. Thanks for reply.  This is just my personal idea.
